### PR TITLE
StatsModels 0.7 compat

### PR DESCRIPTION
**Pull request by palday**
closes #51

It's a big awkward, but we maintain compat with both the 0.6 and 0.7 series.

The CI was copied from https://github.com/beacon-biosignals/StandardizedPredictors.jl/pull/26

### CODE REVIEW

File: src/typical.jl [hunk 1]
```diff
@@ -90,7 +90,14 @@ function _replace(term::InteractionTerm, typicals::Dict)
 end
 
 _termsyms(t) = StatsModels.termsyms(t)
-_termsyms(t::FunctionTerm{Fo,Fa,Names}) where {Fo,Fa,Names} = Names
+
+@static if hasfield(FunctionTerm, :args)
+    # StatsModels 0.7
+    _termsyms(t::FunctionTerm) = StatsModels.termsyms.(only(t.args).args)
```
Comment (kleinschmidt) [inline src/typical.jl:96]: what's happening with the `only` here?  I'd have thought you want 
```suggestion
    _termsyms(t::FunctionTerm) = StatsModels.termsyms.(t.args)
```

Comment (palday) [reply src/typical.jl:96]: I was trying to test locally and might have had a problematic example.

File: src/typical.jl [hunk 2]
```diff
@@ -90,16 +90,22 @@ function _replace(term::InteractionTerm, typicals::Dict)
...
 _symequal(t1::AbstractTerm, t2::AbstractTerm) = issetequal(_termsyms(t1), _termsyms(t2))
 
 _trmequal(t1::AbstractTerm, t2::AbstractTerm) = _symequal(t1, t2)
 _trmequal(t1::AbstractTerm, t2::FunctionTerm) = false
 _trmequal(t1::FunctionTerm, t2::AbstractTerm) = false
 function _trmequal(t1::FunctionTerm, t2::FunctionTerm)
     return t1.exorig == t2.exorig &&
-           _symequal(t1, t2) &&
-           t1.forig == t2.forig
+           _symequal(t1, t2)
```
Comment (palday) [inline src/typical.jl:108]: `forig` went away in 0.7 and it was already a very conservative check, so just dropped it

File: src/typical.jl (hunk 1 above)
Comment (kleinschmidt) [reply src/typical.jl]: oh might need a `reduce(union)` around this...

**Review by kleinschmidt — APPROVED** [review]

**Pull request merged**